apps/condo/domains/user/schema/SignInUserService.js

```diff
@@ -0,0 +1,236 @@
+        code: BAD_USER_INPUT,
+        type: 'DIFFERENT_PHONE_NUMBERS',
+        message: 'The verified phone number and the phone number from the payload cannot be different',
+    },
+    INVALID_PHONE_NUMBER: {
+        mutation: 'signInUser',
+        code: BAD_USER_INPUT,
+        type: 'INVALID_PHONE_NUMBER',
+        message: 'Invalid phone number',
+        messageForUser: 'api.user.signInUser.INVALID_PHONE_NUMBER',
+    },
+    USER_NOT_FOUND: {
+        mutation: 'signInUser',
+        code: NOT_FOUND,
+        type: 'USER_NOT_FOUND',
+        message: 'User not found',
+        messageForUser: 'api.user.signInUser.USER_NOT_FOUND',
+    },
+    TOKEN_NOT_FOUND: {
+        mutation: 'signInUser',
+        variable: ['data', 'token'],
+        code: NOT_FOUND,
+        type: 'TOKEN_NOT_FOUND',
+        message: 'Token not found',
+    },
+    MULTIPLE_USERS_FOUND: {
+        mutation: 'signInUser',
+        code: NOT_FOUND,
+        type: 'MULTIPLE_USERS_FOUND',
+        message: 'Multiple users found',
+    },
+    DV_VERSION_MISMATCH: {
+        query: 'signInUser',
+        variable: ['data', 'dv'],
+        code: BAD_USER_INPUT,
+        type: DV_VERSION_MISMATCH,
+        message: 'Wrong value for data version number',
+    },
+    WRONG_SENDER_FORMAT: {
+        query: 'signInUser',
+        variable: ['data', 'sender'],
+        code: BAD_USER_INPUT,
+        type: WRONG_FORMAT,
+        message: 'Invalid format of "sender" field value',
+        correctExample: '{ dv: 1, fingerprint: \'example-fingerprint-alphanumeric-value\'}',
+    },
+}
+
+const getUserPayload = (user, userData) => {
+    const payload = {}
+    for (const [key, value] of Object.entries(userData))  {
+        if (!value) continue
+        if (get(user, key)) continue
+        payload[key] = value
+    }
+    return payload
+}
+
+const checkRequiredUserFields = (requiredFields, userData) => {
+    if (!Array.isArray(requiredFields)) throw new Error('"requiredFields" should be an array')
+
+    const missingFields = []
+    for (const field of requiredFields) {
+        const value = get(userData, field, null)
+        if (!value && typeof value !== 'boolean' && typeof value !== 'number') missingFields.push(field)
+    }
+    return { missingRequiredFields: missingFields.length > 0, missingFields }
+}
+
+const checkDayRequestLimitCounters = async (context, rawIp) => {
+    const ip = rawIp.split(':').pop()
+    const key = `signInUser:ip:${ip}`
+    const byIpCounter = await REDIS_GUARD.incrementDayCounter(key)
+    if (byIpCounter > MAX_ANONYMOUS_REQUESTS_BY_DAY && !IP_WHITE_LIST.includes(ip)) {
+        throw new GQLError(GQL_ERRORS.SMS_FOR_IP_DAY_LIMIT_REACHED, context)
+    }
+}
+
+const SignInUserService = new GQLCustomSchema('SignInUserService', {
+    types: [
+        {
+            access: true,
+            type: 'input SignInUserUserDataInput { phone: String, name: String, email: String, password: String, meta: JSON }',
```

Comment:
Do we use `meta` in user already? for what reason?
Mb add type for this field?

Reply:
The meta field is currently of type `JSON`. I don't think that should be part of this pr

But yes, it would be nice to add a specific structure for this field